specification/eventgrid/Azure.Messaging.EventGrid/main.tsp

```diff
@@ -0,0 +1,292 @@
+  model PublishResult {}
+
+  @doc("The result of the Acknowledge operation.")
+  model AcknowledgeResult {
+    @doc("Array of LockToken values for failed cloud events. Each LockToken includes the lock token value along with the related error information (namely, the error code and description).")
+    failedLockTokens: FailedLockToken[];
+
+    @doc("Array of lock tokens values for the successfully acknowledged cloud events.")
+    succeededLockTokens: string[];
+  }
+
+  @doc("The result of the Release operation.")
+  model ReleaseResult {
+    @doc("Array of LockToken values for failed cloud events. Each LockToken includes the lock token value along with the related error information (namely, the error code and description).")
+    failedLockTokens: FailedLockToken[];
+
+    @doc("Array of lock tokens values for the successfully released cloud events.")
+    succeededLockTokens: string[];
+  }
+
+  @doc("The result of the Reject operation.")
+  model RejectResult {
+    @doc("Array of LockToken values for failed cloud events. Each LockToken includes the lock token value along with the related error information (namely, the error code and description).")
+    failedLockTokens: FailedLockToken[];
+
+    @doc("Array of lock tokens values for the successfully rejected cloud events.")
+    succeededLockTokens: string[];
+  }
+
+  @doc("Array of lock token strings for the corresponding received Cloud Events to be released.")
+  model ReleaseOptions {
+    @doc("String array of lock tokens.")
+    lockTokens: string[];
+  }
+
+  @doc("Array of lock token strings for the corresponding received Cloud Events to be acknowledged.")
+  model AcknowledgeOptions {
+    @doc("String array of lock tokens.")
+    lockTokens: string[];
+  }
+
+  @doc("Array of lock token strings for the corresponding received Cloud Events to be rejected.")
+  model RejectOptions {
+    @doc("String array of lock tokens.")
+    lockTokens: string[];
+  }
+
+  // Publish Operation:
+  // POST https://{namespaceName}.{region}.eventgrid.azure.net/topics/{topicName}:publish?api-version={apiVersion}}
+
+  @doc("Publish Single Cloud Event to namespace topic. In case of success, the server responds with an HTTP 200 status code with an empty JSON object in response. Otherwise, the server can return various error codes. For example, 401: which indicates authorization failure, 403: which indicates quota exceeded or message is too large, 410: which indicates that specific topic is not found, 400: for bad request, and 500: for internal server error. ")
+  @route("/topics/{topicName}:publish", {shared: true})
+  @post op PublishCloudEvent is Azure.Core.RpcOperation<{
+    @doc("content type")
+    @header("content-type")
+    contentType: "application/cloudevents+json; charset=utf-8";
+
+    @doc("Topic Name.")
+    @path
+    topicName: string;
+
+    @doc("Single Cloud Event being published.")
+    @body 
+    event: CloudEvent;
+  }, PublishResult>;
```

Comment:
Why do we return an empty object? Can we just don't return anything?

Reply:
Is the service really returning {} as the body?

Reply:
The service does return a 200 with '{}'.